fairseq/data/indexed_dataset.py

```diff
@@ -440,11 +448,11 @@ def __len__(self):
 
     def __getitem__(self, i):
         ptr, size = self._index[i]
-        tensor = torch.from_numpy(np.frombuffer(self._bin_buffer, dtype=self._index.dtype, count=size, offset=ptr))
-        if tensor.dtype == torch.int64:
-            return tensor
-        else:
-            return tensor.long()
+        np_array = np.frombuffer(self._bin_buffer, dtype=self._index.dtype, count=size, offset=ptr)
+        if self._index.dtype != np.int64:
+            np_array = np_array.astype(np.int64)
```

Comment:
Does this change improve memory usage and/or speed as well?

Reply:
This is due the fact that invoking `tensor.long()` on a `uint16` tensor was throwing an exception (tensor supports casting only on `int64`, `int32`, `uint8`, ... except `uint16`).
This way I bypass the problem!